### PR TITLE
Adding Jinja2 example

### PR DESCRIPTION
**Pull request by nunogrl**
Documentation: Adding Jinja2 example

Provided example showing how to use `sceptre_user_data` and jinja2.

### CODE REVIEW

File: docs/_source/docs/templates.rst [hunk 1]
```diff
@@ -31,6 +31,121 @@ Template. Sceptre User Data is accessible within Templates as
...
+          Type: A
+          HostedZoneId: !Ref Zone
+          AliasTarget:
+            DNSName: "{{ rule.dnsnamealias }}"
+            HostedZoneId: "{{ rule.hostedzoneid }}"
+      {% endfor %}{% endif %}
+    Outputs:
+      {% if sceptre_user_data.CNAMErecords is defined %}{% for rule in sceptre_user_data.CNAMErecords %}add{{ rule.record |    replace("-","d")|replace("_","s")|replace('.',"p")}}cnamerecord:
+        Value: !Ref 'add{{ rule.record |replace("-","d")|replace("_","s")|replace('.',"p")}}cnamerecord'
+        # Description: 'add{{ rule.record |replace("-","d")|replace("_","s")|replace('.',"p")}}cnamerecord'
```
Comment (ngfgrant) [inline docs/_source/docs/templates.rst]: is this deliberately commented out?

File: docs/_source/docs/templates.rst [hunk 2]
```diff
@@ -31,6 +31,121 @@ Template. Sceptre User Data is accessible within Templates as
...
+        Description: '{{ rule.address }}'
+      {% endfor %}{% endif %}
+      {% if sceptre_user_data.ALIASrecords is defined %}{% for rule in sceptre_user_data.ALIASrecords %}add{{ rule.record |    replace("-","d")|replace("_","s")|replace('.',"p")}}aliasrecord:
+        Value: !Ref 'add{{ rule.record |replace("-","d")|replace("_","s")|replace('.',"p")}}aliasrecord'
+        # Description: 'add{{ rule.record |replace("-","d")|replace("_","s")|replace('.',"p")}}aliasrecord'
```
Comment (ngfgrant) [inline docs/_source/docs/templates.rst]: is this deliberately commented out?

File: docs/_source/docs/templates.rst (hunk 1 above)
Comment (nunogrl) [reply docs/_source/docs/templates.rst]: no, this was a mistake. Thanks for noticing :smile:

File: docs/_source/docs/templates.rst (hunk 2 above)
Comment (nunogrl) [reply docs/_source/docs/templates.rst]: fixed! thanks

File: docs/_source/docs/templates.rst [hunk 3]
```diff
@@ -31,6 +31,119 @@ Template. Sceptre User Data is accessible within Templates as
...
+
+Stack:
+
+.. code-block:: yaml
+
+    template_path: dns-extras.j2
+    dependencies:
+    - prod/route53/domain-zone.yaml
+    parameters:
+      DomainName: "example.com"!stack_output prod/route53/example-com-zone.yaml::FullDomainName
```
Comment (zaro0508) [inline docs/_source/docs/templates.rst]: there's a dependency to `prod/route53/domain-zone.yaml` but it is not provided as part of this example?  should we just remove it?  if this is an example of creating a jinja2 template then i would suggest reducing the scope.

File: docs/_source/docs/templates.rst [hunk 4]
```diff
@@ -31,6 +31,119 @@ Template. Sceptre User Data is accessible within Templates as
...
+      Zone: !stack_output prod/route53/example-com-zone.yaml::HostedZoneID
+    sceptre_user_data:
+      CNAMErecords:
+        - record: "example01"
+          address: "example01.otherdomain.com."
+          ttl: 600
```
Comment (zaro0508) [inline docs/_source/docs/templates.rst]: I think sceptre requires passing in all sceptre numbers in quotes (as strings)?

Comment (nunogrl) [reply docs/_source/docs/templates.rst]: it's working as is :) but I agree that the documentation should present the best practice.

File: docs/_source/docs/templates.rst (hunk 3 above)
Comment (nunogrl) [reply docs/_source/docs/templates.rst]: this contribution is a working stack :) I agree with you. I shall replace the dependency with the values to make it easier to the reader

**Pull request merged**